### PR TITLE
D89: Refactor + Debug

### PR DESCRIPTION
**Pull request by ragnarkurmwunder**
- Refactor the whole core for readability and maintainability.
- Add an option to debug check statuses and time profiling.

Note - there is no point to follow the diff of the PR - just read the whole new file.

I also see the Travis file, and being referred from the readme, is this still relevant?

Note! This is incompatible with the previous version. This means that the composer's `require` has to be mildly different. Therefore need to tag it as 2.0.

### CODE REVIEW

**Comment by ragnarkurmwunder** [comment]
Screenshot 2022-03-24 at 16 15 56

**Comment by tharna** [comment]
Travis is only running some basic syntax checks. Probably doesn't hurt keeping it there.
Should we somehow protect the access to this info. There are some low level information exposed in the output about the underlying stack.

File: test/settings.php [hunk 1]
```diff
@@ -27,14 +27,15 @@
 $elasticsearch = json_decode(getenv('LANDO_INFO'))->elasticsearch;
```
Comment (tormi) [inline test/settings.php:27]: This can be deleted.

File: .lando.yml [hunk 2]
```diff
@@ -18,7 +18,28 @@ services:
...
+        ES_JAVA_OPTS: "-Xms512m -Xmx512m"
+        discovery.type: "single-node"
+        bootstrap.memory_lock: "true"
+      ulimits:
+        memlock:
+          soft: "-1"
+          hard: "-1"
+      ports:
+        - "9200:9200"
+      volumes:
```
Comment (tormi) [inline .lando.yml:38]: Volumes can be removed I believe:

```
      volumes:
        - esdata:/usr/share/elasticsearch/data
    volumes:
      esdata:
        driver: local
```

**Comment by tormi** [comment]
Just FYI - Drupal testing instance doesn't get installed with the latest Composer (`composer_version: 2` in `.lando.yml`) because the new `allow-plugins` config needs manual interference. See also https://www.drupal.org/project/drupal/issues/3255749.

**Comment by tharna** [comment]
One memcache instance being down shouldn't fail the whole test. Same with the ES, but maybe ES shouldn't fail completely anyway in cases where it's only used for search. But on some decoupled solutions it then should (perhaps make it configurable?).

File: .lando.yml (hunk 2 above)
Comment (ragnarkurmwunder) [reply .lando.yml]: thanks!
done.

Comment (ragnarkurmwunder) [reply .lando.yml]: removed the volumes.

**Review by tharna — APPROVED** [review]
Great rewrite. Adds a lot of clarity to the code.

**Pull request merged**